core/src/main/java/jenkins/install/SetupWizard.java

```diff
@@ -55,79 +65,81 @@
 
     private final Jenkins jenkins;
 
-    public SetupWizard(Jenkins j) throws IOException, InterruptedException {
+    public SetupWizard(Jenkins j, boolean newInstall) throws IOException, InterruptedException {
         this.jenkins = j;
 
-        UpgradeWizard.completeUpgrade(j);
-        
-        // Create an admin user by default with a 
-        // difficult password
-        FilePath iapf = getInitialAdminPasswordFile();
-        if(j.getSecurityRealm() == null || j.getSecurityRealm() == SecurityRealm.NO_AUTHENTICATION) { // this seems very fragile
-            BulkChange bc = new BulkChange(j);
-            try{
-                HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(false, false, null);
-                j.setSecurityRealm(securityRealm);
-                String randomUUID = UUID.randomUUID().toString().replace("-", "").toLowerCase(Locale.ENGLISH);
-
-                // create an admin user
-                securityRealm.createAccount(SetupWizard.initialSetupAdminUserName, randomUUID);
-
-                // JENKINS-33599 - write to a file in the jenkins home directory
-                // most native packages of Jenkins creates a machine user account 'jenkins' to run Jenkins,
-                // and use group 'jenkins' for admins. So we allo groups to read this file
-                iapf.touch(System.currentTimeMillis());
-                iapf.chmod(0640);
-                iapf.write(randomUUID + System.lineSeparator(), "UTF-8");
+        if(newInstall) {
```

Comment:
this is the only real change here, moved some of these things to only happen for a new install, rest is whitespace